### DISCUSSION TITLE
Explanation needed: Approval workflow with grouping

### DISCUSSION THREAD

**Discussion by gkranis** [renovatebot/renovate#19042] [Suggest an Idea]
Can someone explain how does grouping affect the approval workflow in the dashboard?

Lets say i group all deps of a repo into a single group. Then let renovate run. Some deps require approval, some do not. 

Will renovatebot offer 2 PRs?
- one for the deps that do not require approval
- the other (needing approval through the dashboard) for the deps that need approval

It seems to me that it instead creates a single PR that requires approval and contains both approval and non-approval requiring deps.

Thanks

**Comment by viceice** [comment]
grouping approvals and non-approvals are not supported. renovate always uses settings from first package of the group.

**Reply by rarkins** [reply]
Actually, we made this smarter some time ago so that PRs require approval if _any_ upgrades in the PR needs approval: https://github.com/renovatebot/renovate/blob/94aac174821f8e8c448f7dfe2df5227c1226adfe/lib/workers/repository/updates/generate.ts#L328-L330

**Reply by gkranis** [reply]
ok so my understanding of the current behaviour is correct, Thanks both
